### PR TITLE
[Docs] Fix documentation building instructions

### PR DESCRIPTION
**Pull request by pcmoritz**
## Why are these changes needed?

It is often a bit challenging to get the full documentation to build (there are external packages that can make this challenging). This changes the instructions to treat warnings as warnings and not errors, which should improve the workflow.

`make develop` is the same as `make html` except it doesn't treat warnings as errors.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by pcmoritz** [comment]
cc @maxpumperla

File: doc/README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ pip install -r requirements-doc.txt
 To compile the documentation and open it locally, run the following command from this directory.
 
 ```bash
-make html && open _build/html/index.html
+make develop && open _build/html/index.html
```
Comment (richardliaw) [inline doc/README.md:19]: Can you also make sure to explicitly say that this is for development, and to reproduce build failures you should still run make html?

File: doc/source/ray-contribute/docs.ipynb [hunk 2]
```diff
@@ -45,7 +45,7 @@
     "Building the documentation is done by running the following command:\n",
     "\n",
     "```shell\n",
-    "make html\n",
+    "make develop\n",
```
Comment (richardliaw) [inline doc/source/ray-contribute/docs.ipynb:48]: same here

File: doc/README.md (hunk 1 above)
Comment (pcmoritz) [reply doc/README.md:19]: added

File: doc/source/ray-contribute/docs.ipynb (hunk 2 above)
Comment (pcmoritz) [reply doc/source/ray-contribute/docs.ipynb:48]: updated

**Review by richardliaw — APPROVED** [review]

**Pull request merged**